lib/internal/buffer.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const Buffer = require('buffer').Buffer;
+const normalizeEncoding = require('internal/util').normalizeEncoding;
+
+if (process.binding('config').hasIntl) {
```

Comment:
what about possibly placing this in `lib/internal/buffer-transcode.js` and conditionally `require()`'ing it. purely cosmetic though, to prevent an extra level of indent. or you could just return early. :)
```js
if (!process.binding('config').hasIntl)
  return;
```